src/main/java/net/snowflake/ingest/streaming/internal/ChannelState.java

```diff
@@ -0,0 +1,59 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/** Internal state of channel that is used and mutated by both channel and its buffer. */
+class ChannelState {
```

Comment:
Maybe rename to `ChannelRuntimeState`? To show that this is an object holding the state of the channel that changes at runtime, contrary to the `ChannelData` which contains the "identity" data about the channel. WDYT?

Reply:
Sure if it is more readable :)